File: project/cs/v1cs/v1cs/Catalog.cs
/***********************************************************************
 * Module:  Catalog.cs
 * Author:  Nadia Abrashina
 * Purpose: Definition of the Class Catalog
 ***********************************************************************/

using System;

public class Catalog
{
   public void AddBook()
   {
      // TODO: implement
   }
   
   public void SearchByName()
   {
      // TODO: implement
   }
   
   public void SearchByPartname()
   {
      // TODO: implement
   }
   
   public void SearchBySubj()
   {
      // TODO: implement
   }
   
   public void LookAll()
   {
      // TODO: implement
   }

   public System.Collections.ArrayList book;
   
   /// <pdGenerated>default getter</pdGenerated>
   public System.Collections.ArrayList GetBook()
   {
      if (book == null)
         book = new System.Collections.ArrayList();
      return book;
   }
   
   /// <pdGenerated>default setter</pdGenerated>
   public void SetBook(System.Collections.ArrayList newBook)
   {
      RemoveAllBook();
      foreach (Book oBook in newBook)
         AddBook(oBook);
   }
   
   /// <pdGenerated>default Add</pdGenerated>
   public void AddBook(Book newBook)
   {
      if (newBook == null)
         return;
      if (this.book == null)
         this.book = new System.Collections.ArrayList();
      if (!this.book.Contains(newBook))
         this.book.Add(newBook);
   }
   
   /// <pdGenerated>default Remove</pdGenerated>
   public void RemoveBook(Book oldBook)
   {
      if (oldBook == null)
         return;
      if (this.book != null)
         if (this.book.Contains(oldBook))
            this.book.Remove(oldBook);
   }
   
   /// <pdGenerated>default removeAll</pdGenerated>
   public void RemoveAllBook()
   {
      if (book != null)
         book.Clear();
   }

   private Book[] Books;

}

File: project/c++/V3/Version1/Version1/Catalog.cpp
/***********************************************************************
 * Module:  Catalog.cpp
 * Author:  Nabra
 * Modified: 19 октября 2016 г. 0:01:24
 * Purpose: Implementation of the class Catalog
 ***********************************************************************/

#include "Book.h"
#include "Catalog.h"
#include <iostream>
#include <vector>

using namespace std;

//=========== Конструктор (тип?) ===========
Catalog::Catalog(const std::vector<Book>&mBooks)
	:mBook(mBooks)
{
	
	cout << "Catalog created by initialisation" << endl;
}

//=========== Конструктор копії ===========
Catalog::Catalog(const Catalog&cat)
{
	mBook = cat.mBook;
	cout << "Catalog created by copy" << endl;
}

//=========== Деструктор =========== 
Catalog::~Catalog()
{
	cout << "Catalog distroed" << endl;
}

//Зміна книги
void Catalog::CBook(int n)
{
	mBook[n].ChangeBook();
}

//Виведення інформації про книгу
void Catalog::GBook(int n)
{
	mBook[n].OutputInformation();
}
////////////////////////////////////////////////////////////////////////
// Name:       Catalog::AddBook()
// Purpose:    Implementation of Catalog::AddBook()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Catalog::AddBook(void)
{
   // TODO : implement
}

//повернути ідентифікатор книги за зазначеним номером
Book& Catalog::getBook(int n)//n - номер книги
{
	return mBook[n];
}

////////////////////////////////////////////////////////////////////////
// Name:       Catalog::SearchByName()
// Purpose:    Implementation of Catalog::SearchByName()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Catalog::SearchByName(void)
{
   // TODO : implement
}

////////////////////////////////////////////////////////////////////////
// Name:       Catalog::SearchByPartname()
// Purpose:    Implementation of Catalog::SearchByPartname()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Catalog::SearchByPartname(void)
{
   // TODO : implement
}

////////////////////////////////////////////////////////////////////////
// Name:       Catalog::SearchBySubj()
// Purpose:    Implementation of Catalog::SearchBySubj()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Catalog::SearchBySubj(void)
{
   // TODO : implement
}

////////////////////////////////////////////////////////////////////////
// Name:       Catalog::LookAll()
// Purpose:    Implementation of Catalog::LookAll()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Catalog::LookAll(void)
{
   // TODO : implement
}